### PR TITLE
776 track outdated browser

### LINKED ISSUE #776: Tracker le nombre connexion depuis des navigateurs non supportés
### Problème 

Comment determine-t-on le nombre de "browser outdated" ?

- compter le nombre de bandeaux outdated
- compter le nombre de page qui chargent react
- compter le nombre de page qui chargent vite

à comparer au total. 

Difficulté : comment connaitre le nombre exact de navigateurs ? Certains bloquent matomo, sont trop vieux ou ont désactivé javascript etc.

Méthodes que l'on peut mettre en oeuvre  : 
- event matomo
- sentry
- pixel de tracking (le seul qui va fonctionner sur les vieux vieux navigateurs)

NB : sur la frugalité, c'est peut etre pas la peine de logguer pour toutes les pages, et en meme temps c'est le plus simple

### PR DESCRIPTION
**Pull request by johangirod**
fix #776

### CODE REVIEW

File: sentry.client.config.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import * as Sentry from '@sentry/nextjs';
 import { isNextJSSentryActivated } from '#utils/sentry';
+declare global {
+  interface Window {
+    IS_OUTDATED_BROWSER: boolean;
+  }
+}
```
Comment (XavierJp) [inline sentry.client.config.ts:7]: I dont see the use for this. As far as we know the whole next js bundle will not load if browser is outdated

The only scenario where sentry might log an erro is if sentry frontend (from the vite bundle)  load independantly from react

File: sentry.client.config.ts [hunk 2]
```diff
@@ -9,5 +14,11 @@ if (isNextJSSentryActivated) {
     // An error can be thrown when a fetch request is aborted during a page unload
     // We don't want to log it to sentry so we ignore it.
     ignoreErrors: ['RequestAbortedDuringUnloadException'],
+    beforeSend(event, hint) {
+      if (window.IS_OUTDATED_BROWSER) {
+        return null;
+      }
+      return event;
+    },
```
Comment (XavierJp) [inline sentry.client.config.ts:22]: Idem I would remove this

File: utils/analytics/matomo.ts
Comment (XavierJp) [inline utils/analytics/matomo.ts:1]: Well done, I think I removed the last call to `logEventInMatomo` in #748

**Review by XavierJp — APPROVED** [review]
I would remove the `window.IS_OUTDATED_BROWSER`.

Do you think it is relevant to add an event for React ? It will allow us to create a double report in matomo : 
- X% user outdated
- Y% user load react

-> we well be able to compare X + Y ~ 100%

 In this case I would change a bit the event.

I would use `browser-support` instead of `error` as the event category, as in matomo events gets grouped by category (https://stats.data.gouv.fr/index.php?module=CoreHome&action=index&idSite=145&period=range&date=previous30#?period=range&date=previous30&idSite=145&category=General_Actions&subcategory=Events_Events)

I would then log `outdated` for this event and `supported` for React event as the event action

File: sentry.client.config.ts (hunk 1 above)
Comment (johangirod) [reply sentry.client.config.ts:7]: TODO à documenter

File: sentry.server.config.ts [hunk 3]
```diff
@@ -13,7 +13,11 @@ if (isNextJSSentryActivated) {
 
       const headers = Object.entries(event.request.headers ?? {}).filter(
         // Remove sensitive headers
-        ([key]) => !key.startsWith('sec-') && !key.startsWith('x-')
+        ([key, value]) =>
+          !value.match(
+            // IP adress regex
+            /(\b25[0-5]|\b2[0-4][0-9]|\b[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}/
+          )
```
Comment (XavierJp) [inline sentry.server.config.ts:20]: good !

**Review by XavierJp — APPROVED** [review]

**Pull request merged**